src/valid/expression.rs

```diff
@@ -633,11 +633,11 @@ impl super::Validator {
                 use crate::UnaryOperator as Uo;
                 let inner = resolver.resolve(expr)?;
                 match (op, inner.scalar_kind()) {
-                    (_, Some(Sk::Sint))
-                    | (_, Some(Sk::Bool))
-                    //TODO: restrict Negate for bools?
```

Comment:
I did not see any reason to properly check all types so that TODO should be fixed by this change.